oci/layout/oci_archive_transport.go

```diff
@@ -0,0 +1,100 @@
+package layout
+
+import (
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/image"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+)
+
+func init() {
+	transports.Register(ArchiveTransport)
+}
+
+// ArchiveTransport is an ImageTransport for OCI archive
+// it creates an oci-archive tar file by calling into the OCI transport
+// tarring the directory created by oci and deleting the directory
+var ArchiveTransport = ociArchiveTransport{}
+
+type ociArchiveTransport struct{}
+
+var ociTr = ociTransport{}
```

Comment:
The `types.ImageTransport` implementations should generally be stateless singletons (containers-storage is the only exception so far). Just call `oci/layout.Transport.$method` directly if necessary.